connection.go

```diff
@@ -344,3 +400,89 @@ func (sc *snowflakeConn) stopHeartBeat() {
 	}
 	sc.rest.HeartBeat.stop()
 }
+
+func updateRows(data execResponseData) (int64, error) {
+	var count int64 = 0
+	for i, n := 0, len(data.RowType); i < n; i++ {
+		v, err := strconv.ParseInt(*data.RowSet[0][i], 10, 64)
+		if err != nil {
+			return -1, err
+		}
+		count += v
+	}
+	return count, nil
+}
+
+type childResult struct {
+	id  string
+	typ string
+}
+
+func getChildResults(IDs string, types string) ([]childResult, error) {
+	if IDs == "" {
+		return nil, nil
+	}
+	queryIDs := strings.Split(IDs, ",")
+	resultTypes := strings.Split(types, ",")
+	if len(queryIDs) != len(resultTypes) {
+		return []childResult{}, &SnowflakeError{
+			Number:  ErrNoChildrenReturned,
+			Message: errMsgFailedToRetrieveChild,
+		}
+	}
```

Comment:
When could this happen? Is it a user or system error? If this is a user error, the error message should be actionable.

Reply:
If this is a system error, dump the debug information to the log and/or include them in the error message so that the users can contact to the support with the information.

Reply:
This would be a system error where the server side didn't return the proper number of children to begin with. I'll add it to the log and error message for the users to see.